src/main/app/src/pages/HomePage/StateChangeResultModal.tsx

```diff
@@ -0,0 +1,198 @@
+  color: ${({ theme }) => theme.colors.red.main};
+`;
+
+const useTableColumns = (t, entityType, getLinkUrl) => [
+  {
+    title: t('yleiset.nimi'),
+    key: 'nimi',
+    render: item => (
+      <NimiLink item={item} entityType={entityType} getLinkUrl={getLinkUrl} />
+    ),
+  },
+  {
+    title: t('etusivu.hakukohde.tilamuutos'),
+    key: 'tilamuutos',
+    render: item => <Tila item={item} entityType={entityType} />,
+  },
+  {
+    title: t('etusivu.hakukohde.tilamuuttunut'),
+    key: 'tilamuuttunut',
+    render: item => {
+      const status = item.status;
+      switch (status) {
+        case 'success':
+          return <SuccessIcon />;
+        case 'error':
+          return <ErrorIcon />;
+        default:
+          return null;
+      }
+    },
+  },
+  {
+    title: t('etusivu.hakukohde.tilaVirhe'),
+    key: 'tilaVirhe',
+    render: item => {
+      const status = item.status;
+      return status === 'error' ? <Virhe errors={item.errorPaths} /> : null;
+    },
+  },
+  {
+    title: t('etusivu.hakukohde.tilaVirheSyy'),
+    key: 'tilaVirheSyy',
+    render: item => {
+      const status = item.status;
+      return status === 'error' ? <Virhe errors={item.errorMessages} /> : null;
+    },
+  },
+];
+
+export const StateChangeResultList = ({ data, entityType, getLinkUrl }) => {
+  const { t } = useTranslation();
+  const columns = useTableColumns(t, entityType, getLinkUrl);
+  const rows = useMemo(
+    () => _fp.map(result => ({ ...result, key: result.oid }), data),
+    [data]
+  );
+
+  return <ListTable rows={rows} columns={columns} />;
+};
+
+type StateChangeResultItem = {
+  oid: string;
+};
+
+const usePreviousNonNil = value => {
+  const prev = usePrevious(value);
+  return _fp.isEmpty(value) ? prev : value;
+};
+
+const isStateChangeResultSuccessful = mutationResult =>
+  _fp.isArray(mutationResult?.data) &&
+  _fp.every(result => result.status === 'success', mutationResult?.data);
+
+export const StateChangeResultModal = ({
+  entityType,
+  headerText,
+  mutationResult,
+  getLinkUrl,
+}: {
+  entityType: ENTITY;
+  headerText: string;
+  mutationResult: UseMutationResult<
+    Array<StateChangeResultItem>,
+    unknown,
+    Array<string>
+  >;
+  getLinkUrl: any;
+}) => {
+  const { t } = useTranslation();
+
+  const isOpen = ['success', 'error'].includes(mutationResult.status);
+
+  const { removeSelection } = useEntitySelection(entityType);
+
+  const queryClient = useQueryClient();
+
+  const { setHakukohdeTila } = useHakukohdeTila();
+
+  const onClose = useCallback(() => {
+    if (isStateChangeResultSuccessful(mutationResult)) {
+      removeSelection();
+      setHakukohdeTila(null);
+    }
+    mutationResult.reset();
+
+    queryClient.invalidateQueries('search.homepage.hakukohteet');
+  }, [mutationResult, removeSelection, queryClient, setHakukohdeTila]);
+
+  const data = usePreviousNonNil(mutationResult.data);
+
+  return (
+    <Modal
+      minHeight="90vh"
+      maxWidth="1200px"
```

Comment:
Näitä arvoja on useammassakin paikassa, voisi jossain vaiheessa tehdä constantin, ehkä myös Modalista jokin oma filunsa

Reply:
Tulosmodaali irroitettu omaksi komponentiksi.